internal/compiler/generator/cpp.rs

```diff
@@ -1411,6 +1409,7 @@ fn generate_item_tree(
         create_code.push("slint::cbindgen_private::slint_ensure_backend();".into());
     }
 
+    let root_access = if parent_ctx.is_some() { "parent->root" } else { "self" };
```

Comment:
Why do we need `parent->root` at all? Doesn't every component have a `root` field?

Reply:
Yes, in order to initialize this->root, we need to take that value from parent->root

Reply:
Right, but that's only needed one single time in the component constructor. Any subsequent access could go through "root" directly. I'm just thinking that this bug is a consequence of "copy & paste" where basically "root_access" was erroneously to loose and could always have been just "root".

Reply:
But I see that my point is moot. In the generator state the root_access is "correct", and this is indeed the last use left for the call to register_component in the constructor.